activity-stream/mofo-eoy-2017.html

```diff
@@ -278,6 +278,16 @@
       var donationURL = "";
       var delimiter = "?";
 
+      btnDonate.addEventListener("click", function(event) {
+          if (event.target.dataset.blocked !== 'true') {
+              event.preventDefault();
+              event.stopPropagation();
+              event.target.dataset.blocked = 'true'
+              addToBlockList();
+          }
+          event.target.click();
+      });
```

Comment:
If the intent is to call `addToBlockList()` before the page navigates elsewhere, I think you could do this a little simpler:

```
event.preventDefault();
event.stopPropagation();
addToBlockList();

// give addToBlockList 250ms to do it's thing before navigating away
setTimeout(function() {
    document.location = event.target.href;
}, 250);
```

I don't think we need any `data-*` attributes set here, as the page will navigate away anyway, thereby wiping any dynamically created `data-*` attributs.

Reply:
Ah, I see. Makes sense. Carry on!